xdsl/parser/attribute_parser.py

```diff
@@ -653,10 +655,17 @@ def _parse_builtin_dense_attr(self, _name: Span) -> DenseIntOrFPElementsAttr:
         # If it is `[]`, then this is a splat attribute, meaning it has the same
         # value everywhere.
         shape: list[int] | None
+        hex_string: str | None = None
         if self._current_token.text == ">":
             values, shape = [], None
         else:
-            values, shape = self._parse_tensor_literal()
+            # Parse hex string e.g. dense<"0x82F5AB00">
+            hex_string = self.parse_optional_str_literal()
+            if hex_string is not None:
+                values, shape = [], None
+            else:
+                # Expect a tensor literal instead
+                values, shape = self._parse_tensor_literal()
```

Comment:
feels like we would benefit from a dedicated helper to parse binary data encoded in a string literal, would make it easier to test, also